src/AppInstallerSharedLib/Yaml.cpp

```diff
@@ -106,11 +106,19 @@ namespace AppInstaller::YAML
             std::string line;
             size_t currentLine = 1;
 
+            // convert BOMs to strings

```

Comment:
BOM should only be the beginning of the document, so the correct way of doing it will be like how we handle it in PrepareInput(), compare the first 2 - 3 bytes and remove them if needed (a new yamlDocumentProcessed string without BOM).

By the way, not your fault, but if BOM is utf16, below code needs to use std::wstring and convert to utf8 string before returning. I remember somewhere we enforced utf-8 for winget-pkgs manifests? If that's the case, this might not be a problem and only utf8 BOM needs to be checked.

Reply:
Thanks for the advice! I'd originally not implemented it there due to the comment in `PrepareInput()` that the BOM was intentionally passed through. However, it seems this was only done so the parser could automatically determine the input encoding.

I've updated the PR to be entirely contained within `PrepareInput()`, removing the BOM if needed (as suggested) and directly setting the parser encoding based on the BOM.